tests/run-pass/sync.rs

```diff
@@ -1,18 +1,140 @@
-// Just instantiate some data structures to make sure we got all their foreign items covered.
-// Requires full MIR on Windows.
+#![feature(rustc_private)]
 
-use std::sync;
+use std::sync::{Mutex, TryLockError};
+
+extern crate libc;
 
 fn main() {
-    let m = sync::Mutex::new(0);
-    drop(m.lock());
+    test_mutex_stdlib();
+    #[cfg(not(target_os = "windows"))] // TODO: implement RwLock on Windows
+    {
+        test_mutex_libc_init_recursive();
+        test_mutex_libc_init_normal();
+        test_rwlock_stdlib();
+        test_rwlock_libc_static_initializer();
+    }
+    #[cfg(target_os = "linux")]
+    {
+        test_mutex_libc_static_initializer_recursive();
```

Comment:
Hm, we already have a `libc.rs`, do you think it would make more sense to move this test there?

Reply:
(Same for all the "libc" tests.)